src/roswire/ros2/state.py

```diff
@@ -28,31 +28,61 @@ class ROS2SystemState(SystemState):
         A mapping from topics to the names of subscribers to that topic.
     services: Mapping[str, Collection[str]]
         A mapping from services to the names of providers of that service.
+    clients: Mapping[str, Collection[str]]
+        A mapping from services to the names of clients of that service.
+    action_servers: Mapping[str, Collection[str]]
+        A mapping from actions to the names of providesr of that action.
+    action_clients: Mapping[str, Collection[str]]
+        A mapping from actions to the names of clients of that action
     nodes: AbstractSet[str]
         The names of all known nodes running on the system.
     topics: AbstractSet[str]
         The names of all known topics on the system with at least one
         publisher or one subscriber.
+    all_services: AbstractSet[str]
```

Comment:
This is a cumbersome and inconsistent name, but I can't call it `services` because that breaks the contract for SystemState.

Reply:
Can you call it `service_clients` to be consistent with `action_clients`?

Reply:
Maybe `service_names` will do for now?

Reply:
Done